app/lib/smaxtec_api.rb

```diff
@@ -5,21 +5,32 @@ class SmaxtecApi
   SMAXTEC_API_PASSWORD = Rails.application.secrets.smaxtec_api_password
   SMAXTEC_API_BASE_URL = 'https://api-staging.smaxtec.com/api/v1'
 
-  def get_temperature
+  def update_sensor_readings
+    Sensor.where.not(animal_id: [nil, ""]).each do |sensor|
+      # temperature sensor
+      if sensor.sensor_type.property == 'Temperature'
+        # get temperature data from smaxtec api
+        temperature = get_temperature(sensor.animal_id)
+
+        if temperature
+          # create new sensor readig for sensor
+          sensor_reading = Sensor::Reading.new(sensor_id: sensor.id, calibrated_value: temperature, uncalibrated_value: temperature)
+          sensor_reading.save!
```

Comment:
If you try to `save` a sensor reading that is already there (ie. some kind of unique attribute is there already) the return value of `save` will be false. In that case maybe you can print out the `sensor_reading.errors` and simple continue on. The procedure will stop this script from creating duplicate data